pkg/eth/tx_listener.go

```diff
@@ -89,11 +96,25 @@ func (t *TxListener) PollForTxCompletion(txID string, updates chan<- string) {
 }
 
 func (t *TxListener) checkTx(hash common.Hash) (bool, error) {
-	_, isPending, err := t.blockchain.TransactionByHash(context.Background(), hash)
+
+	// todo(dankins): TransactionByHash is not implemented by SimulatedBackend
+	// once https://github.com/ethereum/go-ethereum/pull/19026 is resolved we should be able to use the following:
+	// _, isPending, err := t.blockchain.TransactionByHash(context.Background(), hash)
+
+	// this method has no way of determining if a tx just doesn't exist,
+	// so there is potential an invalid tx ID will just poll forever
```

Comment:
If there is concern over this while we wait for the code to be committed to go-ethereum, we could add a timer into `PollForTxCompletion` that fails the loop over some time period.